### ISSUE TITLE
"file" doesn't trickle down to sub_channel children

### ISSUE THREAD

**Issue by codemann8** [qwertymodo/msupcmplusplus#4]
The "file" attribute doesn't properly default to the parent for each sub_channel.  The workaround is to redefine the "file" on each sub_channel separately, but probably is an undesired solution.

Also, when the "file" attribute isn't defined, the program exits and doesn't offer any information/errors within the "verbosity" details.

**Comment by qwertymodo** [comment]
I'm unable to reproduce the sub_channel inheritance problem.  Do you have a reproducible example?  The lack of error when no file is specified is intentional behavior, for a couple of reasons.  For one, I will usually start out a pack by filling in all of the track numbers and titles, and then only add the files as I start working, and it's annoying to get errors for every single track I haven't gotten to yet.  I may look into updating the schema file a bit to see if I can get it to throw a schema error or warning in that case (I think it was already supposed to do that, but I wasn't super familiar with json schemas when I made it).

**Comment by codemann8** [comment]
I remember it was for a case when I had like 6 or so sub-channels, 5 of them using 1 file source, and the other sub-channel using another, so I tried defining the 5 on the root level, thinking it would trickle  down, and then I'd define the different file directly on that one sub-channel.  I'll dig to see if I can find that specific scenario to reproduce it.

**Comment by codemann8** [comment]
Example (removed option 1 and 2 content to condense the example):
```json
	{
		"track_number": 33,
		"title": "Hero",
		"file": "M:\\Music\\_OST\\Naruto OST\\TV.OST-III\\22 - Hero.mp3",
		"use_option": 3,
		"options": [
			{
				"option": 1
			},
			{
				"option": 2
			},
			{
				"option": 3,
				"sub_channels": [
					{
						"pad_start": 11025,
						"sub_tracks": [
							{
								"tempo": 0.92,
								"fade_in": 11025,
								"trim_start": 211388,
								"trim_end": 554893,
								"fade_out": 44100
							}
						]
					},
					{
						"pad_start": 448204,
						"sub_tracks": [
							{
								"tempo": 1.02,
								"fade_in": 22100,
								"trim_start": 813674,
								"trim_end": 6905389
							}
						]
					},
					{
						"pad_start": 6415258,
						"sub_tracks": [
							{
								"tempo": 1.02,
								"fade_in": 5000,
								"trim_start": 7959712,
								"trim_end": 9015563
							}
						]
					},
					{
						"pad_start": 7439479,
						"sub_tracks": [
							{
								"tempo": 1.02,
								"fade_in": 11025,
								"trim_start": 9444394,
								"trim_end": 12266211
							}
						]
					}
				]
			}
		]
        }
```

**Comment by qwertymodo** [comment]
What happens if you move the file line inside of option 3, like this?

```
{
				"option": 3,
				"file": "M:\\Music\\_OST\\Naruto OST\\TV.OST-III\\22 - Hero.mp3",
				"sub_channels": [
. . .
```

**Comment by codemann8** [comment]
Same result... I had to add the file into the next level in on each sub-channel node for it to work

**Comment by qwertymodo** [comment]
Fixed in 89368c4cb98a1b8d249ace0096ec8f6fd24cd45e
